### PR TITLE
fix: timestamping

### PR DESCRIPTION
**Pull request by Two-Hearts**
Due to implementation needs, we must provide a `time.Time` when verifying TSA certificate chain validity after signing. Originally, this time value was taken from the timestamp countersignature envelope itself. 
This PR switches to use the `current time` when verifying TSA certificate chain validity since the tsa cert chain is expected to be valid during signing.

### CODE REVIEW

**Review by JeyJeyGao — APPROVED** [review]
LGTM

File: internal/timestamp/timestamp.go [hunk 1]
```diff
@@ -43,16 +43,8 @@ func Timestamp(req *signature.SignRequest, opts tspclient.RequestOptions) ([]byt
...
-	if err != nil {
-		return nil, err
-	}
-	timestamp, err := info.Validate(opts.Content)
-	if err != nil {
-		return nil, err
-	}
 	tsaCertChain, err := token.Verify(ctx, x509.VerifyOptions{
-		CurrentTime: timestamp.Value,
+		CurrentTime: req.SigningTime,
```
Comment (shizhMSFT) [inline internal/timestamp/timestamp.go:47]: We should remain the previous implementation due to security consideration we have discussed offline.

Besides, we should have a sanity check on the `req.SigningTime` and the `timestamp.Value` as it only makes sense if the `req.SigningTime` is after the `timestamp.Value`.

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: internal/timestamp/timestamp.go (hunk 1 above)
Comment (Two-Hearts) [reply internal/timestamp/timestamp.go]: Updated.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Comment by priteshbandi** [comment]
>This PR switches to use the current time when verifying TSA certificate chain validity since the tsa cert chain is expected to be valid during signing.

Is this true for verification ?

**Comment by Two-Hearts** [comment]
> Is this true for verification ?

This PR is for signing only. We will have another PR for verification in notation-go once this PR gets merged.

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**